qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/IqtFunctionTemplateModel.cpp

```diff
@@ -0,0 +1,351 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "IqtFunctionTemplateModel.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidQtWidgets/Common/FunctionBrowser/FunctionBrowserUtils.h"
+
+#include <map>
+#include <tuple>
+
+namespace {
+using namespace MantidQt::CustomInterfaces::IDA;
+
+std::tuple<double, double> calculateLifetimeAndHeight(Mantid::MantidVec const &x, Mantid::MantidVec const &y) {
+  auto lifeTime = (x[1] - x[0]) / (log(y[0]) - log(y[1]));
+  if (lifeTime <= 0 || std::isinf(lifeTime))
```

Comment:
Perhaps we can avoid computing the infinite, 
```suggestion
  auto lifeTime= ((x[0]>=x[1]) || (y[0]==y[1])) ? 1.0 : (x[1] - x[0]) / (log(y[0]) - log(y[1]));
```

Reply:
I've added a check to make sure the denominator is not zero. It is a bit more difficult to check the lifetime is not going to be below zero before doing the calculation, so I have left this in an if statement